### ISSUE TITLE
Reboots at [ PCI configuration begin ]

### ISSUE THREAD

**Issue by fearandloathing** [RevoGirl/RevoBoot#28]
I try, I try, and I try (sounds like a Queen song) but I can never seem to get past this point. Is there any way to figure out what is causing it to reboot here. This is the farthest that I have gotten and prior to this I would always get a KP on AppleACPIPlatformExpert::start failed "Unable to find driver for this platform".

**Comment by RevoGirl** [comment]
Rebooting is, in my experience, a problem with the compiler directives. Like it compiles /boot for Snow Leopard instead of  (Mountain) Lion. But then usually very early in the boot process. You seem to get past this stage so it has to be something different.

Are you using dynamic CPU and SMBIOS data gathering, because using the wrong SMBIOS data is also known to trigger a reboot. Even more so than the compiler directives.

I am currently very busy – trying to get past the security error in the 3K Asus UEFI-BIOS – and may not find the time to look into this problem, but I will as soon I have a moment of time (came back here, the next day, to add more text in the hope to clarify things for you).

**Comment by fearandloathing** [comment]
Hi Samantha, thanks for the reply, I am in no hurry so whenever you
get a chance, and good luck with the security error (I'm sure you'll
figure it out). Just for reference when I compile  I use "make clean"
and "make lion" I have tried using the different compiler directives
but those reboot before I even get console output. Also if I run a
continuous ping to my IP address when I boot I actually get like 8
icmp echo replies at the point where it says [PCI configuration begin]
then it reboots. That makes me think I am so close to getting this to
work since it has to be OS X sending the icmp echo replies.
Kind regards,
Dan

On Wed, Apr 25, 2012 at 9:00 AM, Samantha
reply@reply.github.com
wrote:

**Comment by RevoGirl** [comment]
Hi Dan,

That is interesting. I mean that it pings. 

What happens when you rename /S_/L_/E*/AppleIntelCPUPowerManagement.kext?
AppleHDA.kext would be the next suspect, but I still think that this is another SMBIOS problem.

**Comment by fearandloathing** [comment]
I reboot at the same point if I remove that kext. 

If I add -s to the caBp Kernel Flags instead of a reboot my machine hangs after [ PCI configuration begin ]. Though keyboard does work (Caps lock key) but no video or network so I can't tell if it is just sitting there (potentially at the OS X login prompt) or if it really is hung.

Does it matter that I am using a Gigabyte GA-Z68X-UD3H-B3?

On Thu, Apr 26, 2012 at 1:30 PM, Samantha
reply@reply.github.com
wrote:

**Comment by RevoGirl** [comment]
I checked my notes. Had troubles with the "PCI configuration begin" message myself on a HP notebook. The problem was resolved when I fixed the DSDT, which was really broken. On my hack I don't even use a custom DSDT anymore. I also don't use any kernel flags anymore.

AUTOMATIC_SSDT_PR_CREATION is either 3 or 7 depending on the board/UEFI-BIOS I am using (latest 3K doesn't need SBUS injection anymore).

LOAD_SSDT_TABLE_FROM_EXTRA_ACPI is set to 1 simply because I want to load a SSDT.aml with device name changes in it and to fix wake after sleep. Boot fine without it.

The other compiler directives are pretty much the default.

And no. I don't think that using Gigabyte has anything to do with it.

p.s. I think to have the Xcode 4.3 (and greater) compiler troubles covered. 

Update: The latest update (in the tree) fixed the compiler problems for Xcode 4.3.1 (boots fine now).

**Comment by fearandloathing** [comment]
Nice work on the Xcode fix.

Pretty basic DSDT with the standard renames etc. I did try using one from TMAC as well with the same results. That said I did notice the same problem with Chimera which was fixed using the Kernel Flag npci=0x3000. Is that flag an option with Revoboot.

**Comment by RevoGirl** [comment]
Thank you.

Are you using my stripped DSDT now? I ask this, because I don't see the need for it. That is. Not when the ACPI tables are patched correctly.

About the kernel flag. That is what it is: XNU kernel flag. Had nothing to do with Chimera, Chameleon or RevoBoot. Can be used on gal Mac's also.

**Comment by fearandloathing** [comment]
Nope, no stripped DSDT. I have tried using no DSDT at all, using ACPI tables that I exported from Revobuilder etc. 
The XNU kernel flag helps and now I can see that I don't actually get stuck at PCI Config, it is actually KP'ing with the following Kernel Extensions in backtrace.
com.apple.driver.ApplyPolicyControl......
I'm sure its just something stupid that I am missing and I am sure that it doesn't help that I don't know much about programming. I'm gonna educate myself (downloaded the Harvard CS50 podcast) and learn how to do some programming. Been reading the ACPI spec but man it's cryptic for someone that doesn't know programming.
Don't waste anymore time on me as I'm sure it's my lack of knowledge that is the problem.
One thing I am not is a quitter and I am going to get this to work if it's the last thing I do.
Thanks for all of your help, I'll be back.
